### ISSUE TITLE
Load scripts from subdirectory of config/scripts - compiles, but not found on execution.

### ISSUE THREAD

**Issue by twigbranch** [elastic/elasticsearch#12353]
I've placed in my config/scripts directory a symlink to another directory containing a script. It appears that elasticsearch found and compiled the script, as it states "compiling script file," in the elasticsearch.log.

When I try to access it in a bulk update, I get "ElasticsearchIllegalArgumentException[Unable to find on disk script ..."

**Comment by twigbranch** [comment]
I tried copying the file to the root of config/scripts and got the same "compiling script file." But when attempting to execute it in a bulk update, got the same illegal argument exception as above.

Are there some other requirements to execute the script? Or is it now confused because it has compiled a .groovy with the same name twice?

**Comment by twigbranch** [comment]
The problem is I specified the script_file to include the file extension, apparently you can only specify the base name.  Not sure why.

**Comment by colings86** [comment]
Please stick to asking these types of questions on the http://discuss.elastic.co forums. This issues list is used for bug and new feature tracking. Also, posting in multiple places can often lead to separate disjointed conversation which can be hard to follow not only for the people involved now but also for people trying to answer the same question later.

**Comment by twigbranch** [comment]
This appears to be a bug. it says the script compiled, but it won't execute.  

if it isn't a bug then the documentation needs to be fixed.

**Comment by colings86** [comment]
Apologies, you also have this raised in the discuss forums (https://discuss.elastic.co/t/script-file-in-symlinked-subdirectory-of-config-scripts-not-executing/26016) as a question and I was trying to keep the conversation in one place so it can be followed easily. The error you quoted above is truncated, could you paste the whole error message (ideally the whole response for the single index operation in the bulk) here as well as the corresponding part of the bulk request for that index operation?

**Comment by twigbranch** [comment]
Thanks- my apologies, I had not seen any response to that post (and had checked it a few minutes prior to finding the closure).  Here is the request response for a single 'update' (it is the same error for a bulk update):

```
/myapp/release_path/config/elasticsearch/scripts/ includes:
myscript_name.groovy
```
```
ln -nfs /myapp/release_path/config/elasticsearch/scripts /etc/elasticsearch/scripts/myapp
```

Elasticsearch.log:
```
compiling script file [/etc/elasticsearch/scripts/myapp/myscript_name.groovy]
```

Request:
```
curl -XPOST 'localhost:9200/test/type1/1/_update' -d '{
     "script" : {
          "lang": "groovy",
          "script_file": "myscript_name",
          "params": {
               operation: false,
               new_doc: true
          }
     }
}'
```
Response:
```
Elasticsearch::Transport::Transport::Errors::BadRequest: [400] {"error":"ElasticsearchIllegalArgumentException[failed to execute script]; nested: ElasticsearchIllegalArgumentException[Unable to find on disk script myscript_name]; ","status":400}
```

**Comment by twigbranch** [comment]
I also tried physically copying the script into a subdirectory of /etc/elasticsearch/scripts.  As above, elasticsearch claims to find and compile the script, but upon attempt to execute, returns the 'unable to find' error.

I also tried symlinking to a script file in /etc/elasticsearch/scripts (as opposed to symlinking the directory), and I do not see a "compiling script file" in the log for that symlinked file.

So, it appears, elasticsearch finds and compiles files in a symlinked subdirectory, but it isn't clear how to access and execute them.  And, elasticsearch does not find and/or compile symlinked files in the config/scripts directory itself.
